rsocket/internal/ManageableConnection.h

```diff
@@ -38,6 +38,13 @@ class ManageableConnection {
   virtual void close(folly::exception_wrapper,
                      StreamCompletionSignal) = 0;
 
+  /// If the connection is already disconnected, then there is
+  /// no need to close it in the given eventbase, we can call
+  /// close() function inline.
```

Comment:
Is that truly safe?  We don't depend on being on the right EventBase thread in the stream state machine methods that get called from `RSocketStateMachine::closeStreams()`?

Reply:
This can be too risky to do.  We have a lot of callbacks in close() and the application might have some expectation on it.  I think a hard failure here is better (fail fast).  It is better to set the expectation that the evb which drives the RSocket should outlive the instances.